session.go

```diff
@@ -1095,9 +1086,8 @@ func (s *session) prepareTxnCtx() {
 		return
 	}
 
-	txnFuture := s.getTxnFuture()
-	goCtx, cancelFunc := goctx.WithCancel(goctx.Background())
-	s.txnFuture, s.goCtx, s.cancelFunc = txnFuture, goCtx, cancelFunc
+	s.goCtx, s.cancelFunc = goctx.WithCancel(goctx.Background())
```

Comment:
For historical reasons.
Each transaction goes along with a new context, transaction run in a session,
so the context is stored in the session.